Sources/_SubprocessCShims/include/process_shims.h

```diff
@@ -35,11 +35,18 @@
 #include <sys/wait.h>
 #endif // TARGET_OS_LINUX || TARGET_OS_FREEBSD
 
+#ifdef __cplusplus
+#define SWIFTSUBPROCESS_EXPORT extern "C"
+#else
+#define SWIFTSUBPROCESS_EXPORT extern
+#endif
+
```

Comment:
I wonder if this would be better served with a:

```c++
extern "C" {
```

block rather than annotating each of the declarations.

Reply:
That's what I originally had, and then I saw `#include <Windows.h>` and realized I couldn't just wrap the entire thing with `extern "C"`.

... But on second thought, I could just hoist that up to the top, and do the `extern "C"` thing you suggest (which is a cleaner patch).

Reply:
Right, as long as the includes come outside of the block, it is simpler. That also avoids the pitfall of missing the annotation on a future declaration.